### DISCUSSION TITLE
useFeeData always returns mainnet fee data on first load

### DISCUSSION THREAD

**Discussion by lsbyerley** [wevm/wagmi#130] [Q&A] [answered]
I am using the `useFeeData` hook for displaying a module with the estimated gas price. The fee data returned on initial load is always the eth mainnet gas price even if the wallet is on a testnet such as rinkeby. Is this expected?

```jsx

const [{ data: feeData, error: feeError, loading: feeLoading }, getFeeData] =
  useFeeData({ formatUnits: 'gwei', skip: true });

useEffect(async () => {
  // this always returns eth mainnet gas price when the component first mounts
  await getFeeData();
}, []);

useInterval(
  async () => {
    // this is an interval that runs every 10 seconds to update the fee data and it actually does return the current network's gas price
    await getFeeData();
  },
  accountAddress ? 10000 : null
);

**Comment by lsbyerley** [comment, marked as answer]
It appears using the `watch: true` flag for the useFeeData hook resolves this.. apologies for not seeing that option farther down in the docs

**Reply by tmm** [reply]
No worries! Check out [this comment](https://github.com/tmm/wagmi/discussions/111#discussioncomment-2086092).
